### PR TITLE
Nano: drop frozen V10 config, make triggerObjectTable more easily customizable

### PR DESCRIPTION
**Pull request by swertz**
#### PR description:

- Drop the frozen nanoV10 configuration introduced in https://github.com/cms-sw/cmssw/pull/39337 as we do not plan to support v10 in 13_0_X
   - Also remove v10 workflows from the matrix
   - Simplify the triggerObjectTableProducer as backward-compatibility with the v10 config is no longer needed
- Make the triggerObjectTable more easily customizable with eras/modifers by making the object selection a `PSet` instead of a `VPSet`
- Add back Run2-specific jet trigger object filters for Run2 only, removed in https://github.com/cms-sw/cmssw/pull/39818
    - This required removing the `run2_HLTconditions_2018` modifier from the `Run3` era. From a quick search I don't expect this to have unwanted effects elsewhere (note that this relates to the discussion in https://github.com/cms-sw/cmssw/issues/39958 as this is not the first time we've needed to introduce a different behaviour between Run2 and Run3 for things that were blindly inherited from the `Run2_2018` era to the `Run3` era, see e.g. https://github.com/cms-sw/cmssw/pull/40209). 
    - @lathomas  FYI, it meant that the L1 prefiring weights applied in nanoAODv10 to Run3 samples were taken as the 2018UL ones (and still are in nanoV11 in 12_6_X), which is probably incorrect.

#### PR validation:

Ran nano matrix workflows.

### CODE REVIEW

**Comment by swertz** [comment]
enable nano

**Comment by swertz** [comment]
please test

**Comment by swertz** [comment]
Looking at the jet trigger filters for Run2
- 2016: 
image

- 2017: 
image

- 2018: 
image


It seems that the "Run3" trigger filters introduced in https://github.com/cms-sw/cmssw/pull/39818 (therefore appearing in blue) are actually *more* relevant than the "Run2" filters previously present in UL nanoV9 (appearing in red), in particular for 2017 and 2018 menus.

I think for completeness we should make sure all the necessary Run2 filters are present, so I'll make further changes to add some of the filters from #39818 also for Run2.

@johnalison does that make sense to you?

**Comment by johnalison** [comment]
@swertz Makes sense to me as long as there are spare bits to use. If we run out, I guess its best to have different configurations for the different eras.

**Comment by swertz** [comment]
please test

@johnalison we're actually nearly out of bits (unless we move to `uint64_t`), so I've used era-dependent tweaks based on what I could make out from QCD MC samples.

**Comment by swertz** [comment]
please test

**Comment by swertz** [comment]
+1

Expected changes in Run2 jet trigger object filter bits (and more objects present as a result); L1prefiring weights removed from Run3 samples.

**Comment by srimanob** [comment]
+Upgrade

From the upgrade related-code, only clean up in `Configuration/PyReleaseValidation`.

**Comment by swertz** [comment]
@cms-sw/operations-l2 @cms-sw/pdmv-l2 do you have any comments on this?

**Comment by kskovpen** [comment]
+pdmv

**Comment by perrotta** [comment]
please test
(This seems to touch also non-nanoAOD workflows: let have the three weeks old test refreshed first, and then we should try to understand it)

File: Configuration/Eras/python/Era_Run3_cff.py [hunk 1]
```diff
@@ -12,8 +12,9 @@
 from Configuration.ProcessModifiers.dd4hep_cff import dd4hep
 from Configuration.Eras.Modifier_run3_egamma_cff import run3_egamma
 from Configuration.Eras.Modifier_run2_egamma_2018_cff import run2_egamma_2018
+from Configuration.Eras.Modifier_run2_HLTconditions_2018_cff import run2_HLTconditions_2018
 from Configuration.Eras.Modifier_run3_RPC_cff import run3_RPC
 
-Run3 = cms.ModifierChain(Run2_2018.copyAndExclude([run2_GEM_2017, ctpps_2018, run2_egamma_2018]),
+Run3 = cms.ModifierChain(Run2_2018.copyAndExclude([run2_GEM_2017, ctpps_2018, run2_egamma_2018, run2_HLTconditions_2018]),
```
Comment (perrotta) [inline Configuration/Eras/python/Era_Run3_cff.py:18]: @missirol @Martin-Grunewald  could you please confirm that this is ok for HLT in Run3?

Comment (missirol) [reply Configuration/Eras/python/Era_Run3_cff.py:18]: > could you please confirm that this is ok for HLT in Run3?

Yes, it is okay.

To my knowledge, nothing that concerns HLT directly depends on this modifier. It is used in the HLT Offline DQM (FYI: @cms-sw/dqm-l2 (@ckoraka)), e.g.
https://github.com/cms-sw/cmssw/blob/a9b836229a8b2869e3d342b70dd1332e45e0681a/DQMOffline/Trigger/python/TopMonitoring_cff.py

For reference, this change was already discussed in https://github.com/cms-sw/cmssw/pull/39082 and https://github.com/cms-sw/cmssw/pull/39084 (neither of which was merged, despite what the GitHub labels say).

Comment (swertz) [reply Configuration/Eras/python/Era_Run3_cff.py:18]: Thanks, looking at the discussion in https://github.com/cms-sw/cmssw/pull/39084 it looks like people actually wanted this change to be included then; not clear why it was reverted in https://github.com/cms-sw/cmssw/pull/39119...

**Comment by perrotta** [comment]
+1
- Also to me it is not clear why the change in https://github.com/cms-sw/cmssw/pull/40321/files#diff-bdce97c132f2495fedca6f8c58fe085659f25973341a407fff38d404b804c9bcR18 was first implemented and then reverted in #39119. I can suspect some mis-usage of github functionalities that silently removed #39084 from the radar. However, everybody agrees that it is welcome for Run3:  let have it implemented here, then.

**Comment by smuzaffar** [comment]
@swertz,  many workflows in latest 13.0.X are failing due to [a]. Looks like this PR has deleted the `PhysicsTools/NanoAOD/V10` directory. Can you please look in to this and provide a fix? thanks

[a]
```
raceback (most recent call last):
  File "/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02767/el8_amd64_gcc11/cms/cmssw/CMSSW_13_0_X_2023-01-11-0800/bin/el8_amd64_gcc11/cmsDriver.py", line 51, in <module>
    run()
  File "/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02767/el8_amd64_gcc11/cms/cmssw/CMSSW_13_0_X_2023-01-11-0800/bin/el8_amd64_gcc11/cmsDriver.py", line 28, in run
    configBuilder.prepare()
  File "/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02767/el8_amd64_gcc11/cms/cmssw/CMSSW_13_0_X_2023-01-11-0800/src/Configuration/Applications/python/ConfigBuilder.py", line 2165, in prepare
    self.addStandardSequences()
  File "/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02767/el8_amd64_gcc11/cms/cmssw/CMSSW_13_0_X_2023-01-11-0800/src/Configuration/Applications/python/ConfigBuilder.py", line 790, in addStandardSequences
    getattr(self,"prepare_"+stepName)(stepSpec = '+'.join(stepSpec))
  File "/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02767/el8_amd64_gcc11/cms/cmssw/CMSSW_13_0_X_2023-01-11-0800/src/Configuration/Applications/python/ConfigBuilder.py", line 1724, in prepare_NANO
    _,_nanoSeq,_nanoCff = self.loadDefaultOrSpecifiedCFF(stepSpec,self.NANODefaultCFF,self.NANODefaultSeq)
  File "/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02767/el8_amd64_gcc11/cms/cmssw/CMSSW_13_0_X_2023-01-11-0800/src/Configuration/Applications/python/ConfigBuilder.py", line 1236, in loadDefaultOrSpecifiedCFF
    l=self.loadAndRemember(_cff)
  File "/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02767/el8_amd64_gcc11/cms/cmssw/CMSSW_13_0_X_2023-01-11-0800/src/Configuration/Applications/python/ConfigBuilder.py", line 331, in loadAndRemember
    self.process.load(includeFile)
  File "/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02767/el8_amd64_gcc11/cms/cmssw/CMSSW_13_0_X_2023-01-11-0800/src/FWCore/ParameterSet/python/Config.py", line 757, in load
    module = __import__(moduleName)
ModuleNotFoundError: No module named 'PhysicsTools.NanoAOD.V10'
```

**Comment by swertz** [comment]
Indeed, that directory is removed and I removed the associated nano-specific workflows from https://github.com/cms-sw/cmssw/blob/master/Configuration/PyReleaseValidation/python/relval_nano.py but I didn't notice that this step was making use of it: https://github.com/cms-sw/cmssw/blob/master/Configuration/PyReleaseValidation/python/relval_steps.py#L2800

How come this didn't come up in the PR tests?

I can make a PR ASAP to switch that step to use the default nano configuration instead of the V10 one.

**Comment by srimanob** [comment]
We don't have the 2022 Data in runTheMatrix,
https://github.com/cms-sw/cmssw/blob/master/Configuration/PyReleaseValidation/scripts/runTheMatrix.py

Maybe we should add one, i.e. 140.106 for JetHT 2022B.

**Comment by smuzaffar** [comment]
PR test runs a selected number of workflows and I guess non of those workflows were using PhysicsTools/NanoAOD/V10/nano_cff

**Pull request merged**